### PR TITLE
Fixes #138

### LINKED ISSUE #138: Cover more code flags in disassembly output
## Description

For the moment I'm observing the case when pydisasm output leaves some flags as hex values,
but it'd be nice to decode it into user friendly strings

## Priority

Low

## Tests

version used:
>pydisasm --version
pydisasm, version 6.1.2.dev0

example of the sample .pyc:

```
# Flags:             0x0100001f (0x1000000 | NESTED | VARKEYWORDS | VARARGS | NEWLOCALS | OPTIMIZED)

```

expected output:
0x1000000 decoded into FUTURE_ANNOTATIONS
as the module is using 

from __future__ import annotations

Thanks!

Attached sample .pyc:
[rewriter.cpython-311.zip](https://github.com/user-attachments/files/16522014/rewriter.cpython-311.zip)

### PR DESCRIPTION
**Pull request by greenozon**
Minor changes to address the issue #138

### CODE REVIEW

File: xdis/__init__.py [hunk 1]
```diff
@@ -119,6 +119,7 @@
     CO_FUTURE_BARRY_AS_DBFL,
     CO_FUTURE_DIVISION,
     CO_FUTURE_GENERATOR_STOP,
+    CO_FUTURE_ANNOTATIONS,
```
Comment (rocky) [inline xdis/__init__.py:122]: Previously these lines were in alphabetic order. Preserving that would be nice.

File: xdis/__init__.py [hunk 2]
```diff
@@ -163,6 +164,7 @@
     "CO_FUTURE_BARRY_AS_DBFL",
     "CO_FUTURE_DIVISION",
     "CO_FUTURE_GENERATOR_STOP",
+    "CO_FUTURE_ANNOTATIONS",
```
Comment (rocky) [inline xdis/__init__.py:167]: See above regarding sorting.

**Comment by rocky** [comment]
Thanks for starting this. The CI tests indicate more work is needed. 

Please also consider writing a CI test. See for example https://github.com/rocky/python-xdis/blob/master/pytest/test_disasm.py#L96-L110

File: xdis/__init__.py (hunk 1 above)
Comment (greenozon) [reply xdis/__init__.py:122]: oh, thanks! I was thinking like a minute deducing what was the order about...  but didn't guess it :)

**Comment by greenozon** [comment]
@rocky  have a quick question on tests:

if I run the tests using python - it passes, but 
if I run using `make check`  as mentioned in the readme - it fails (strange error Access Denied on temp folder..)
Windows env, Python 3.12.5
any clues why is that? tried to run under regular user as well as Administrator - same effect

1)
```
c:\Dev\python\python-xdis-my\python-xdis>python -m pytest pytest
========================================================================================= test session starts =========================================================================================
platform win32 -- Python 3.12.5, pytest-8.3.2, pluggy-1.5.0
rootdir: c:\Dev\python\python-xdis-my\python-xdis
configfile: pyproject.toml
collected 32 items

pytest\test_bytecode.py ..                                                                                                                                                                       [  6%]
pytest\test_codetype.py .                                                                                                                                                                        [  9%]
pytest\test_cross_dis.py .                                                                                                                                                                       [ 12%]
pytest\test_disasm.py ssssssss                                                                                                                                                                   [ 37%]
pytest\test_instructions.py ..                                                                                                                                                                   [ 43%]
pytest\test_load_file.py s                                                                                                                                                                       [ 46%]
pytest\test_magic.py .                                                                                                                                                                           [ 50%]
pytest\test_opcode.py .                                                                                                                                                                          [ 53%]
pytest\test_stack_effect.py ..                                                                                                                                                                   [ 59%]
pytest\test_std.py ............s                                                                                                                                                                 [100%]

========================================================================================== warnings summary ===========================================================================================
pytest/test_codetype.py::test_codeType2Portable
  c:\Dev\python\python-xdis-my\python-xdis\xdis\codetype\__init__.py:44: DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
    line_table_field = "co_lnotab" if hasattr(code, "co_lnotab") else "co_linetable"

pytest/test_codetype.py::test_codeType2Portable
  c:\Dev\python\python-xdis-my\python-xdis\xdis\codetype\__init__.py:45: DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
    line_table = getattr(code, line_table_field)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================= 22 passed, 10 skipped, 2 warnings in 0.28s ==============================================================================

```

2)
```
c:\Dev\python\python-xdis-my\python-xdis>make --version
GNU Make 4.4.1
Built for x86_64-pc-msys
Copyright (C) 1988-2023 Free Software Foundation, Inc.
License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>
This is free software: you are free to change and redistribute it.
There is NO WARRANTY, to the extent permitted by law.

c:\Dev\python\python-xdis-my\python-xdis>make check
python -m pytest pytest
========================================================================================= test session starts =========================================================================================
platform win32 -- Python 3.12.5, pytest-8.3.2, pluggy-1.5.0
rootdir: c:\Dev\python\python-xdis-my\python-xdis
configfile: pyproject.toml
collected 32 items

pytest\test_bytecode.py ..                                                                                                                                                                       [  6%]
pytest\test_codetype.py .                                                                                                                                                                        [  9%]
pytest\test_cross_dis.py .                                                                                                                                                                       [ 12%]
pytest\test_disasm.py ssssssss                                                                                                                                                                   [ 37%]
pytest\test_instructions.py ..                                                                                                                                                                   [ 43%]
pytest\test_load_file.py s                                                                                                                                                                       [ 46%]
pytest\test_magic.py .                                                                                                                                                                           [ 50%]
pytest\test_opcode.py .                                                                                                                                                                          [ 53%]
pytest\test_stack_effect.py ..                                                                                                                                                                   [ 59%]
pytest\test_std.py ............s                                                                                                                                                                 [100%]

========================================================================================== warnings summary ===========================================================================================
pytest/test_codetype.py::test_codeType2Portable
  c:\Dev\python\python-xdis-my\python-xdis\xdis\codetype\__init__.py:44: DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
    line_table_field = "co_lnotab" if hasattr(code, "co_lnotab") else "co_linetable"

pytest/test_codetype.py::test_codeType2Portable
  c:\Dev\python\python-xdis-my\python-xdis\xdis\codetype\__init__.py:45: DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
    line_table = getattr(code, line_table_field)

-- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
============================================================================= 22 passed, 10 skipped, 2 warnings in 0.27s ==============================================================================
make[1]: Entering directory '/c/Dev/python/python-xdis-my/python-xdis/test'
python test_pyenvlib.py --simple
Wed Aug 14 12:29:50 2024
Processed 24 files, total
Wed Aug 14 12:29:50 2024
0.00600004 seconds
Wed Aug 14 12:29:50 2024
Traceback (most recent call last):
  File "C:\Dev\python\python-xdis-my\python-xdis\test\test_pyenvlib.py", line 266, in <module>
    do_tests(
  File "C:\Dev\python\python-xdis-my\python-xdis\test\test_pyenvlib.py", line 185, in do_tests
    check_bc_file = check_object_path(bc_file)
                    ^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "C:\Users\user\AppData\Local\Programs\Python\Python312\Lib\site-packages\xdis\load.py", line 88, in check_object_path
    py_compile.compile(spath, cfile=path, doraise=True)
  File "C:\Users\user\AppData\Local\Programs\Python\Python312\Lib\py_compile.py", line 172, in compile
    importlib._bootstrap_external._write_atomic(cfile, bytecode, mode)
  File "<frozen importlib._bootstrap_external>", line 208, in _write_atomic
PermissionError: [WinError 5] Access is denied: 'C:\\Users\\user\\AppData\\Local\\Temp\\06_frozenset-q2zv43zi.pyc.21511088' -> 'C:\\Users\\user\\AppData\\Local\\Temp\\06_frozenset-q2zv43zi.pyc'
make[1]: *** [Makefile:28: check-short] Error 1
make[1]: Leaving directory '/c/Dev/python/python-xdis-my/python-xdis/test'
make: *** [Makefile:37: check] Error 2

c:\Dev\python\python-xdis-my\python-xdis>
c:\Dev\python\python-xdis-my\python-xdis>python -V
Python 3.12.5

```

**Comment by greenozon** [comment]
working on adding TC for new flag

**Comment by rocky** [comment]
> @rocky have a quick question on tests:
> 
> if I run the tests using python - it passes, but if I run using `make check` as mentioned in the readme - it fails (strange error Access Denied on temp folder..) Windows env, Python 3.12.5 any clues why is that? tried to run under regular user as well as Administrator - same effect
> 

What are the permissions on both the folder and the file?  That is not shown below.

> 
> ```
> c:\Dev\python\python-xdis-my\python-xdis>python -m pytest pytest
> ========================================================================================= test session starts =========================================================================================
> platform win32 -- Python 3.12.5, pytest-8.3.2, pluggy-1.5.0
> rootdir: c:\Dev\python\python-xdis-my\python-xdis
> configfile: pyproject.toml
> collected 32 items
> 
> pytest\test_bytecode.py ..                                                                                                                                                                       [  6%]
> pytest\test_codetype.py .                                                                                                                                                                        [  9%]
> pytest\test_cross_dis.py .                                                                                                                                                                       [ 12%]
> pytest\test_disasm.py ssssssss                                                                                                                                                                   [ 37%]
> pytest\test_instructions.py ..                                                                                                                                                                   [ 43%]
> pytest\test_load_file.py s                                                                                                                                                                       [ 46%]
> pytest\test_magic.py .                                                                                                                                                                           [ 50%]
> pytest\test_opcode.py .                                                                                                                                                                          [ 53%]
> pytest\test_stack_effect.py ..                                                                                                                                                                   [ 59%]
> pytest\test_std.py ............s                                                                                                                                                                 [100%]
> 
> ========================================================================================== warnings summary ===========================================================================================
> pytest/test_codetype.py::test_codeType2Portable
>   c:\Dev\python\python-xdis-my\python-xdis\xdis\codetype\__init__.py:44: DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
>     line_table_field = "co_lnotab" if hasattr(code, "co_lnotab") else "co_linetable"
> 
> pytest/test_codetype.py::test_codeType2Portable
>   c:\Dev\python\python-xdis-my\python-xdis\xdis\codetype\__init__.py:45: DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
>     line_table = getattr(code, line_table_field)
> 
> -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
> ============================================================================= 22 passed, 10 skipped, 2 warnings in 0.28s ==============================================================================
> ```
> 
> 
> ```
> c:\Dev\python\python-xdis-my\python-xdis>make --version
> GNU Make 4.4.1
> Built for x86_64-pc-msys
> Copyright (C) 1988-2023 Free Software Foundation, Inc.
> License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>
> This is free software: you are free to change and redistribute it.
> There is NO WARRANTY, to the extent permitted by law.
> 
> c:\Dev\python\python-xdis-my\python-xdis>make check
> python -m pytest pytest
> ========================================================================================= test session starts =========================================================================================
> platform win32 -- Python 3.12.5, pytest-8.3.2, pluggy-1.5.0
> rootdir: c:\Dev\python\python-xdis-my\python-xdis
> configfile: pyproject.toml
> collected 32 items
> 
> pytest\test_bytecode.py ..                                                                                                                                                                       [  6%]
> pytest\test_codetype.py .                                                                                                                                                                        [  9%]
> pytest\test_cross_dis.py .                                                                                                                                                                       [ 12%]
> pytest\test_disasm.py ssssssss                                                                                                                                                                   [ 37%]
> pytest\test_instructions.py ..                                                                                                                                                                   [ 43%]
> pytest\test_load_file.py s                                                                                                                                                                       [ 46%]
> pytest\test_magic.py .                                                                                                                                                                           [ 50%]
> pytest\test_opcode.py .                                                                                                                                                                          [ 53%]
> pytest\test_stack_effect.py ..                                                                                                                                                                   [ 59%]
> pytest\test_std.py ............s                                                                                                                                                                 [100%]
> 
> ========================================================================================== warnings summary ===========================================================================================
> pytest/test_codetype.py::test_codeType2Portable
>   c:\Dev\python\python-xdis-my\python-xdis\xdis\codetype\__init__.py:44: DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
>     line_table_field = "co_lnotab" if hasattr(code, "co_lnotab") else "co_linetable"
> 
> pytest/test_codetype.py::test_codeType2Portable
>   c:\Dev\python\python-xdis-my\python-xdis\xdis\codetype\__init__.py:45: DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
>     line_table = getattr(code, line_table_field)
> 
> -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
> ============================================================================= 22 passed, 10 skipped, 2 warnings in 0.27s ==============================================================================
> make[1]: Entering directory '/c/Dev/python/python-xdis-my/python-xdis/test'
> python test_pyenvlib.py --simple
> Wed Aug 14 12:29:50 2024
> Processed 24 files, total
> Wed Aug 14 12:29:50 2024
> 0.00600004 seconds
> Wed Aug 14 12:29:50 2024
> Traceback (most recent call last):
>   File "C:\Dev\python\python-xdis-my\python-xdis\test\test_pyenvlib.py", line 266, in <module>
>     do_tests(
>   File "C:\Dev\python\python-xdis-my\python-xdis\test\test_pyenvlib.py", line 185, in do_tests
>     check_bc_file = check_object_path(bc_file)
>                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
>   File "C:\Users\user\AppData\Local\Programs\Python\Python312\Lib\site-packages\xdis\load.py", line 88, in check_object_path
>     py_compile.compile(spath, cfile=path, doraise=True)
>   File "C:\Users\user\AppData\Local\Programs\Python\Python312\Lib\py_compile.py", line 172, in compile
>     importlib._bootstrap_external._write_atomic(cfile, bytecode, mode)
>   File "<frozen importlib._bootstrap_external>", line 208, in _write_atomic
> PermissionError: [WinError 5] Access is denied: 'C:\\Users\\user\\AppData\\Local\\Temp\\06_frozenset-q2zv43zi.pyc.21511088' -> 'C:\\Users\\user\\AppData\\Local\\Temp\\06_frozenset-q2zv43zi.pyc'
> make[1]: *** [Makefile:28: check-short] Error 1
> make[1]: Leaving directory '/c/Dev/python/python-xdis-my/python-xdis/test'
> make: *** [Makefile:37: check] Error 2
> 
> c:\Dev\python\python-xdis-my\python-xdis>
> c:\Dev\python\python-xdis-my\python-xdis>python -V
> Python 3.12.5
> ```

**Comment by rocky** [comment]
Looks better now, although a test would be nice.

**Comment by rocky** [comment]
> > @rocky have a quick question on tests:
> > if I run the tests using python - it passes, but if I run using `make check` as mentioned in the readme - it fails (strange error Access Denied on temp folder..) Windows env, Python 3.12.5 any clues why is that? tried to run under regular user as well as Administrator - same effect
> 
> What are the permissions on both the folder and the file? That is not shown below.
> 
> > ```
> > c:\Dev\python\python-xdis-my\python-xdis>python -m pytest pytest
> > ========================================================================================= test session starts =========================================================================================
> > platform win32 -- Python 3.12.5, pytest-8.3.2, pluggy-1.5.0
> > rootdir: c:\Dev\python\python-xdis-my\python-xdis
> > configfile: pyproject.toml
> > collected 32 items
> > 
> > pytest\test_bytecode.py ..                                                                                                                                                                       [  6%]
> > pytest\test_codetype.py .                                                                                                                                                                        [  9%]
> > pytest\test_cross_dis.py .                                                                                                                                                                       [ 12%]
> > pytest\test_disasm.py ssssssss                                                                                                                                                                   [ 37%]
> > pytest\test_instructions.py ..                                                                                                                                                                   [ 43%]
> > pytest\test_load_file.py s                                                                                                                                                                       [ 46%]
> > pytest\test_magic.py .                                                                                                                                                                           [ 50%]
> > pytest\test_opcode.py .                                                                                                                                                                          [ 53%]
> > pytest\test_stack_effect.py ..                                                                                                                                                                   [ 59%]
> > pytest\test_std.py ............s                                                                                                                                                                 [100%]
> > 
> > ========================================================================================== warnings summary ===========================================================================================
> > pytest/test_codetype.py::test_codeType2Portable
> >   c:\Dev\python\python-xdis-my\python-xdis\xdis\codetype\__init__.py:44: DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
> >     line_table_field = "co_lnotab" if hasattr(code, "co_lnotab") else "co_linetable"
> > 
> > pytest/test_codetype.py::test_codeType2Portable
> >   c:\Dev\python\python-xdis-my\python-xdis\xdis\codetype\__init__.py:45: DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
> >     line_table = getattr(code, line_table_field)
> > 
> > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
> > ============================================================================= 22 passed, 10 skipped, 2 warnings in 0.28s ==============================================================================
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > ```
> > c:\Dev\python\python-xdis-my\python-xdis>make --version
> > GNU Make 4.4.1
> > Built for x86_64-pc-msys
> > Copyright (C) 1988-2023 Free Software Foundation, Inc.
> > License GPLv3+: GNU GPL version 3 or later <https://gnu.org/licenses/gpl.html>
> > This is free software: you are free to change and redistribute it.
> > There is NO WARRANTY, to the extent permitted by law.
> > 
> > c:\Dev\python\python-xdis-my\python-xdis>make check
> > python -m pytest pytest
> > ========================================================================================= test session starts =========================================================================================
> > platform win32 -- Python 3.12.5, pytest-8.3.2, pluggy-1.5.0
> > rootdir: c:\Dev\python\python-xdis-my\python-xdis
> > configfile: pyproject.toml
> > collected 32 items
> > 
> > pytest\test_bytecode.py ..                                                                                                                                                                       [  6%]
> > pytest\test_codetype.py .                                                                                                                                                                        [  9%]
> > pytest\test_cross_dis.py .                                                                                                                                                                       [ 12%]
> > pytest\test_disasm.py ssssssss                                                                                                                                                                   [ 37%]
> > pytest\test_instructions.py ..                                                                                                                                                                   [ 43%]
> > pytest\test_load_file.py s                                                                                                                                                                       [ 46%]
> > pytest\test_magic.py .                                                                                                                                                                           [ 50%]
> > pytest\test_opcode.py .                                                                                                                                                                          [ 53%]
> > pytest\test_stack_effect.py ..                                                                                                                                                                   [ 59%]
> > pytest\test_std.py ............s                                                                                                                                                                 [100%]
> > 
> > ========================================================================================== warnings summary ===========================================================================================
> > pytest/test_codetype.py::test_codeType2Portable
> >   c:\Dev\python\python-xdis-my\python-xdis\xdis\codetype\__init__.py:44: DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
> >     line_table_field = "co_lnotab" if hasattr(code, "co_lnotab") else "co_linetable"
> > 
> > pytest/test_codetype.py::test_codeType2Portable
> >   c:\Dev\python\python-xdis-my\python-xdis\xdis\codetype\__init__.py:45: DeprecationWarning: co_lnotab is deprecated, use co_lines instead.
> >     line_table = getattr(code, line_table_field)
> > 
> > -- Docs: https://docs.pytest.org/en/stable/how-to/capture-warnings.html
> > ============================================================================= 22 passed, 10 skipped, 2 warnings in 0.27s ==============================================================================
> > make[1]: Entering directory '/c/Dev/python/python-xdis-my/python-xdis/test'
> > python test_pyenvlib.py --simple
> > Wed Aug 14 12:29:50 2024
> > Processed 24 files, total
> > Wed Aug 14 12:29:50 2024
> > 0.00600004 seconds
> > Wed Aug 14 12:29:50 2024
> > Traceback (most recent call last):
> >   File "C:\Dev\python\python-xdis-my\python-xdis\test\test_pyenvlib.py", line 266, in <module>
> >     do_tests(
> >   File "C:\Dev\python\python-xdis-my\python-xdis\test\test_pyenvlib.py", line 185, in do_tests
> >     check_bc_file = check_object_path(bc_file)
> >                     ^^^^^^^^^^^^^^^^^^^^^^^^^^
> >   File "C:\Users\user\AppData\Local\Programs\Python\Python312\Lib\site-packages\xdis\load.py", line 88, in check_object_path
> >     py_compile.compile(spath, cfile=path, doraise=True)
> >   File "C:\Users\user\AppData\Local\Programs\Python\Python312\Lib\py_compile.py", line 172, in compile
> >     importlib._bootstrap_external._write_atomic(cfile, bytecode, mode)
> >   File "<frozen importlib._bootstrap_external>", line 208, in _write_atomic
> > PermissionError: [WinError 5] Access is denied: 'C:\\Users\\user\\AppData\\Local\\Temp\\06_frozenset-q2zv43zi.pyc.21511088' -> 'C:\\Users\\user\\AppData\\Local\\Temp\\06_frozenset-q2zv43zi.pyc'
> > make[1]: *** [Makefile:28: check-short] Error 1
> > make[1]: Leaving directory '/c/Dev/python/python-xdis-my/python-xdis/test'
> > make: *** [Makefile:37: check] Error 2
> > 
> > c:\Dev\python\python-xdis-my\python-xdis>
> > c:\Dev\python\python-xdis-my\python-xdis>python -V
> > Python 3.12.5
> > ```

I am seeing the same thing happen in https://github.com/rocky/python-xdis/actions/runs/10390365060/job/28770653197 . I haven't a clue as to why this is happening.

**Comment by greenozon** [comment]
great! a new detective story just begins!
:)
in my case the temp folder having my current user as the owner, but I hit the Access denied even under Admin!
also, those temp .pyc files are of 0 length in size

**Pull request merged**